apollo-federation/src/operation/optimize.rs

```diff
@@ -1579,6 +1589,155 @@ impl Operation {
     }
 }
 
+/// Returns a consistent GraphQL name for the given index.
+fn fragment_name(mut index: usize) -> Name {
```

Comment:
Unsure how much we gain by auto generating random fragment names vs something human readable. Guess folks generally shouldn't be looking at those ops... but it would make it easier to debug if needed.

Reply:
You *always* need something like this to make sure that names don't collide, that decision is just between serializing the number as a string or serializing it to a minimal name like this. Making it human readable is just doing a bit more work ;)
The JS uses a name like `generated_onTypeCondition`, but the type condition is already in the definition, so it doesn't really help in my opinion.